### ISSUE TITLE
C2X conditional operator and qualifiers on pointer to arrays

### ISSUE THREAD

**Issue by uecker** [llvm/llvm-project#54568]
With N2607 arrays are considered to have the same qualifier as the element. Clang mostly
already behaves according to these rules but there seems to be one remaining inconsistency
in the conditional operator.  The standard says:

— both operands are pointers to qualified or unqualified versions of compatible types;

Clang incorrectly warns about a pointer type mismatch in the following example that
should be ok when considering arrays to be qualified according to N2607.

int a[3] = { 0 };
const int b[3] = { 0 };

const void* foo(void)
{
    return (0 ? &a : &b);
}

https://godbolt.org/z/E18rdqa5j
(BTW: This would also remove an inconsistency with respect to C++)

**Comment by AaronBallman** [comment]
Here's a slightly reduced example that shows where we're wrong:
```
int main(void) {
  int a[3];
  const int b[3];

  _Static_assert(__builtin_types_compatible_p(__typeof__(0 ? &a : &b), const int(*)[3]));
}
```
https://godbolt.org/z/PraY7ohze

GCC accepts this, Clang rejects because the type from the conditional operator is `void *`. I suspect the issue lays somewhere in or near `ASTContext::mergeTypes()`.
